docs/build_docs.py

```diff
@@ -0,0 +1,880 @@
+    output_dir : str
+        The directory where the final built documentation will be placed.
+    log : logging.Logger, optional
+        Logger for logging messages; defaults to the module logger.
+    """
+    log.info("Setting docs files up for building")
+    # copy and validate files
+    stage_docs_files_for_building(
+        os.path.join(repo_dir, "docs", "source"),
+        package_name,
+        geoips_docs_dir,
+        build_dir,
+        log=log,
+    )
+
+    # grab auxillary files not in docs and place them in "import" dir
+    import_non_docs_files(repo_dir, build_dir)
+
+    # build release rst files
+    log.info("Building API docs")
+    releases_dir = os.path.join(build_dir, "releases")
+    build_release_notes_with_brassy(releases_dir, license_url)
+
+    # build api doc rst files
+    apidoc_build_path = os.path.join(build_dir, f"{package_name}_api")
+    module_path = os.path.join(repo_dir, package_name)  # module path
+    build_module_apidocs_with_sphinx(module_path, apidoc_build_path, log=log)
+
+    # build final html files
+    with tempfile.TemporaryDirectory() as built_dir:
+        log.info("Building docs")
+        build_docs_with_sphinx(build_dir, built_dir, log=log)
+        log.debug("Docs built successfully")
+        if os.path.exists(output_dir):
+            log.info(
+                f"Removing output directory {output_dir} in preparation of writing"
+                + "built docs"
+            )
+            shutil.rmtree(output_dir)
+        shutil.copytree(built_dir, output_dir, dirs_exist_ok=True)
+
+
+def main(
+    repo_dir,
+    package_name,
+    geoips_docs_dir,
+    output_dir,
+    license_url,
+):
+    """Prepare for and execute documentation build.
+
+    This function initializes logging, validates paths and package installation,
+    logs the configuration, and runs the documentation build process within a temporary
+    directory to avoid cluttering the filesystem.
+
+    Parameters
+    ----------
+    repo_dir : str
+        The root directory of the package's repository.
+    package_name : str
+        The name of the package to build documentation for.
+    geoips_docs_dir : str
+        The path to the GeoIPS documentation directory (usually geoips/docs)
+    output_dir : str
+        The directory where the built documentation will be placed.
+    """
+    log = init_logger(True)
+    log.debug("Program initialized")
+
+    validate_package_is_installed(package_name, logger=log)
+
+    for path in (
+        repo_dir,
+        geoips_docs_dir,
+        os.path.join(repo_dir, "docs"),
+        os.path.join(repo_dir, package_name),
+    ):
+        validate_path_exists_and_is_git_repo(path, logger=log)
+
+    log.info(f"Repo path is {repo_dir}")
+    log.info(f"Repo docs path is {os.path.join(repo_dir, 'docs')}")
+    log.info(f"GeoIPS docs path is {geoips_docs_dir}")
+    log.info(f"Package name is {package_name}")
+
+    # generate release notes here, TODO
```

Comment:
Does this still need to be completed or is this an out of date comment?

Reply:
Fixed 😅 thanks!